api/proto/profile/local_profile.proto

```diff
@@ -54,3 +54,44 @@ message RadioConfig {
    bool radio_silence = 2;
 }
 
+// AppInstances message is sent by the Local Profile Server in response to a GET
+// request issued by EVE to the api/v1/application_instances API.
+message AppInstances {
+   // Security token. EVE will verify that server_token matches the profile server
+   // token received from the controller.
+   string server_token = 1;
+   // Each item in the list should reference a different running application instance.
+   // Entries that do not match any currently running application on the device
+   // will be ignored.
+   repeated AppInstance app_instances = 2;
+}
+
+// AppInstance references a running application instance by UUID and/or displayname,
+// and encapsulates all command/config requests submitted locally for the instance.
+message AppInstance {
+   // Reference the application instance by UUID.
+   // At least one of the UUID and displayname should be defined.
+   string uuid = 1;
+   // Reference the application instance by the user-friendly displayname.
+   // At least one of the UUID and displayname should be defined.
+   string displayname  = 2;
+   // Locally request to restart the referenced application instance.
+   AppInstanceCmd restart = 3;
+   // Locally request to purge the referenced application instance.
+   AppInstanceCmd purge = 4;
+}
+
+// AppInstanceCmd wraps all parameters for an application instance command.
+message AppInstanceCmd {
+   // Increase the counter to trigger the command.
+   // Increasing this multiple times does not imply the command will run
+   // more than once, especially if increments were done at short intervals.
+   // EVE does not persist retrieved counter values. Local Profile Server
+   // should start with zero counters on each restart (which may or may not
+   // be due to a device reboot).
```

Comment:
Small question. If we increase restart counter for appN (1), than for some reason restart Local Profile Server (to update, or to be in online mode...), will we expect 1 or 2 for appN to restart it again?

In any case seems reasonable to document this logic in `api/PROFILE.md`.

Reply:
Actually now that I'm rethinking this, without requiring the profile server to persist the counter on its side (and also eve to do the same), the counter-based solution will not be able to reliably cover the scenarios with server restarts.
A more robust approach would be probably to use timestamps. The profile server would record the time when a purge/restart request for a particular app was made. A zero timestamp would mean that no request has been made since the last server restart. A change between the previous and a new (non-zero) timestamp would be a signal for eve to trigger application purge/restart. The clock of the server app would not need to be in-sync with the device or with a real clock, eve would compare server's timestamps only between themselves, not against its/real time.
@giggsoff Do you see any potential risks with a timestamp-based solution? I know that timekeeping in VMs is tricky and the VM clock could be quite off, but it shouldn't matter here since we would only be comparing successive timestamps made by the same clock.
I will mark this as WIP and will come back with a re-worked design. I will also document this workflow in `api/PROFILE.md` as you suggested.